Mathlib/LinearAlgebra/InvariantBasisNumber.lean

```diff
@@ -76,6 +98,57 @@ section
 
 variable (R : Type u) [Semiring R]
 
+/-- We say that `R` satisfies the Orzech property, if for any finitely generated `R`-module `M`,
+    any surjective homomorphism `f : N → M` from a submodule `N` of `M` to `M` is injective. -/
+@[mk_iff]
+class OrzechProperty : Prop where
+  injective_of_surjective_of_submodule' : ∀ {M : Type u} [AddCommMonoid M] [Module R M]
+    [Module.Finite R M] {N : Submodule R M} (f : N →ₗ[R] M), Surjective f → Injective f
+
+variable {R} in
```

Comment:
I really appreciate that you get rid of the universes constraint.

Reply:
Maybe I should add a docstring to the definition of `OrzechProperty` since I can't use universe polymorphic in the definition. But we actually can prove a universe polymorphic version `OrzechProperty.injective_of_surjective_of_injective` using `small_of_surjective`.

Reply:
Yes, the docstring should say that, and the universe polymorphic version as well.

Reply:
Done.